plugins/core/IO/qLASIO/src/LasTiler.cpp

```diff
@@ -0,0 +1,208 @@
+    } else {
+        pointCount = laszipHeader->number_of_point_records;
+    }
+
+    laszip_point *laszipPoint{nullptr};
+    if (laszip_get_point_pointer(laszipReader, &laszipPoint)) {
+        laszip_get_error(laszipHeader, &errorMsg);
+        ccLog::Warning("[LAS] laszip error: '%s'", errorMsg);
+        laszip_close_reader(laszipReader);
+        laszip_clean(laszipReader);
+        laszip_destroy(laszipReader);
+        return CC_FERR_THIRD_PARTY_LIB_FAILURE;
+    }
+
+    std::vector<laszip_POINTER> writers;
+    writers.resize(options.numTiles0 * options.numTiles1, nullptr);
+
+    CC_FILE_ERROR error = CC_FERR_NO_ERROR;
+
+    const size_t index0 = options.index0();
+    const size_t index1 = options.index1();
+
+    double cloudMins[3] = {
+            laszipHeader->min_x,
+            laszipHeader->min_y,
+            laszipHeader->min_z,
+    };
+
+    double cloudBBox[3] = {
+            laszipHeader->max_x - laszipHeader->min_x,
+            laszipHeader->max_y - laszipHeader->min_y,
+            laszipHeader->max_z - laszipHeader->min_z,
+    };
+
+    double tileSize[2] = {
+            cloudBBox[index0] / static_cast<double>(options.numTiles0),
+            cloudBBox[index1] / static_cast<double>(options.numTiles1),
+    };
+
+    laszip_F64 laszipCoordinates[3] = {0};
+
+    QFileInfo fileInfo(originName);
+    const std::string baseName = fileInfo.baseName().toStdString();
+    const std::string extension = fileInfo.suffix().toStdString();
+
+    QElapsedTimer timer;
+    timer.start();
+
+    ccProgressDialog progressDialog(true);
+    progressDialog.setMethodTitle("Tiling LAS file");
+    progressDialog.setInfo("Tiling...");
+    CCCoreLib::NormalizedProgress normProgress(&progressDialog, pointCount);
+    progressDialog.start();
+
+    for (unsigned i = 0; i < pointCount; ++i) {
+        if (laszip_read_point(laszipReader)) {
+            laszip_get_error(laszipReader, &errorMsg);
+            ccLog::Warning("[LAS] laszip error :'%s'", errorMsg);
+            error = CC_FERR_THIRD_PARTY_LIB_FAILURE;
+            break;
+        }
+
+        if (laszip_get_coordinates(laszipReader, laszipCoordinates)) {
+            laszip_get_error(laszipReader, &errorMsg);
+            ccLog::Warning("[LAS] laszip error :'%s'", errorMsg);
+            error = CC_FERR_THIRD_PARTY_LIB_FAILURE;
+            break;
+        }
+
+        auto tileI = static_cast<size_t>((laszipCoordinates[index0] - cloudMins[index0]) / tileSize[0]);
+        tileI = std::min(tileI, static_cast<size_t>(options.numTiles0 - 1));
+        auto tileJ = static_cast<size_t>((laszipCoordinates[index1] - cloudMins[index1]) / tileSize[1]);
+        tileJ = std::min(tileJ, static_cast<size_t>(options.numTiles1 - 1));
+
+        const size_t tileIndex = (tileI * options.numTiles1) + tileJ;
+
+        laszip_POINTER &writer = writers[tileIndex];
+
+        if (writer == nullptr) {
+            std::string outputName;
```

Comment:
Wouldn't it be simpler to use QStrings to do this and just convert it to stdString when needed? (and baseName and extension can remain QStrings)
if (...)
{
    outputName = options.outputDir + '/';
}
outputName += baseName + '\_' + QString::number(tileI) + '\_'  + QString::number(tileK) + '.' + extension;

(or equivalent)

Reply:
And you can probably retrieve baseName and extension from fileInfo just here (as you don't need them elsewhere)

Reply:
I did it that way to avoid too many allocation from QString to std::string, but I can change that

Reply:
Yes, it's a bit 'heavy' for something that simple